File: src/test/java/de/lukaseichler/recurrentneuralnetwork/structure/InputLayerTest.java
package de.lukaseichler.recurrentneuralnetwork.structure;

import org.testng.annotations.Test;
import com.google.common.collect.Lists;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author leichler
 */
public class InputLayerTest {
    @Test
    public void calculate() throws Exception {
        Layer layer = new InputLayer(2);
        assertThat(layer.calculate(Lists.newArrayList(1.0, 2.0))).containsOnly(1.0, 2.0);
    }
}

File: src/test/scala/de/lukaseichler/recurrentneuralnetwork/structure/InputLayerTest.scala
package de.lukaseichler.recurrentneuralnetwork.structure

import org.scalatest.Matchers
import org.scalatest.testng.TestNGSuite
import org.testng.annotations.Test

/**
 * @author Lukas Eichler
 */
class InputLayerTest extends TestNGSuite with Matchers {

  @Test
  def calculate {
    val layer: Layer = new InputLayer(2)
    layer.calculate(List(1.0, 2.0)) should contain only(1.0, 2.0)
  }
}
